<?xml version="1.0" encoding="UTF-8" standalone="no" ?>
<launch>


  <arg name="robotID" default="1"/>
  <arg name="NUM_ROBOTS" default="2" />


  <node name="minion_nmpc_planner_$(arg robotID)" pkg="minion_nmpc_planner" type="antipodal" args="$(arg robotID) $(arg NUM_ROBOTS)" output="screen">
    <param name="robotPoseTopic" value="/minion_" type="string" />
    <param name="robotPoseTopicSuffix" value="/ground_truth_pose" type="string" />

    <param name="robotPoseTrajTopic" value="/minion_" type="string" />
    <param name="robotPoseTrajTopicSuffix" value="/mpc_calculated/pose_traj" type="string" />

    <param name="outputPoseTopic" value="/waypoint_minion_" type="string" />
    <param name="robotSelfIMUTopic" value="/minion_$(arg robotID)/imu" type="string" />

    <param name="objectGTTopic" value="/actorpose" type="string" />
<!--     <param name="objectGTTopic" value="/virtual_target_square_motion" type="string" />  -->
    <param name="obstacleTopicBase" value="/NewObstaclePoints_$(arg robotID)" type="string" />

    <param name="useGTforTarget" value="true" type="bool" />
    <param name="usingSimulation" value="true" type="bool" />

    <!-- This delta T is for the planner timesteps and should be roughly equal to inverse of self pose frequency -->
    <param name="deltaT" value="0.1" type="double" />
    <!-- This INTERNAL_SUB_STEP is for the potential field timesteps and should be roughly equal to inverse of self pose frequency -->
    <param name="INTERNAL_SUB_STEP" value="0.1" type="double" />

    <param name="tFlyToDestinationRadius" value="0.25" type="double" />
    <param name="tFormationRepulGain" value="30.0" type="double" />
    <param name="tGoalAttractionGain" value="5.0" type="double" />

    <param name="maxObstacleRepulsionForce" value="0.7" type="double" />

    <param name="neighborDistThreshold" value="0.5" type="double" />
    <param name="distanceThresholdToTarget" value="4.0" type="double" />
    <param name="copterDesiredHeightinNED" value="-5.0" type="double" />

    <param name="useZeroAsFixedTarget" value="false" type="bool" />

   </node>
   <!--node name = "plotter_$(arg robotID)" pkg="nmpc_planner" type="plotter.py" output="screen" args="$(arg robotID)"> </node-->

</launch>
